File: api_examples/SimpleApp.scala
import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.functions._

object WordCountApp {
  def main(args: Array[String]): Unit = {
    // Initialize SparkSession
    val spark = SparkSession.builder()
      .appName("WordCount")
      .master("local[*]") 
      .getOrCreate()

    // Data path
    val filePath = "../data/le_petit_prince.txt"

    // Read text file into DataFrame
    val df = spark.read.text(filePath)

    // Word count logic
    val wordCounts = df
      .select(explode(split(lower(col("value")), "\\s+")).alias("word"))
      .filter(col("word") =!= "")
      .groupBy("word")
      .agg(count("*").alias("count"))
      .orderBy(desc("count"))

    // Show top 20 most frequent words
    wordCounts.show(20)

    // Save results
    val outputPath = "../data/out/le_petit_prince_count.txt"
    wordCounts.write.option("header", "true").csv(outputPath)

    spark.stop()
  }
}

File: api_examples/SimpleApp.py
from pyspark.sql import SparkSession
from pyspark.sql.functions import explode, split, lower, col, count


def main():
    # Initialize a Spark Session
    spark = SparkSession.builder \
        .appName("WordCount") \
        .getOrCreate()

    # data path
    file_path = "../data/le_petit_prince.txt"
    # Read text file into DataFrame
    # Each line becomes a row in the DataFrame
    df = spark.read.text(file_path)

    # Process the text and count words
    word_counts = df \
        .select(explode(split(lower(col("value")), "\\s+")).alias("word")) \
        .filter(col("word") != "") \
        .groupBy("word") \
        .count() \
        .orderBy("count", ascending=False)

    # Show top 20 most frequent words
    word_counts.show(20)

    # Save results if needed
    output_file_path = "../data/out/le_petit_prince_count.txt"
    word_counts.write.csv(output_file_path, header=True)

    spark.stop()


if __name__ == "__main__":
    main()